features/working_with_legacy_code/message_chains.feature

```diff
@@ -19,6 +20,12 @@ Feature: Message Chains
   ```ruby
   allow(Article).to receive_message_chain("recent.published") { [Article.new] }
   ```
+  
+  `receive_message_chain` is designed to be used with evaluating a response like `and_return`, 
+  `and_yield` etc. For legacy reasons, parity with `stub_chain` is supported but its uses are
+  not considered good practice. Support for `stub_chain` pairity may be removed in future versions.
+  
+  Common method chains like `with` and `exactly` (i.e. `exactly(2).times`) are not supported.
```

Comment:
We refer to these as customisations internally, and I think this is confusing to call them chains here when we're talking about the chain matcher, chaining here refers to the target not how we set up the expectation.